pvlib/modelchain.py

```diff
@@ -994,7 +998,11 @@ def _set_celltemp(self, model):
                                   self.results.effective_irradiance)
         temp_air = _tuple_from_dfs(self.weather, 'temp_air')
         wind_speed = _tuple_from_dfs(self.weather, 'wind_speed')
-        self.results.cell_temperature = model(poa, temp_air, wind_speed)
+        arg_list = [poa, temp_air, wind_speed]
+        kwargs = {}
+        if model == self.system.noct_sam_celltemp:
+            kwargs['effective_irradiance'] = self.results.effective_irradiance
+        self.results.cell_temperature = model(*tuple(arg_list))
```

Comment:
@cwhanse when attempting to merge your changes into #1197 I realized that this is not correct because kwargs is not passed to the function. I had suggested 

```python
self.results.cell_temperature = model(poa, temp_air, wind_speed, **kwargs)
```

if you want to keep `arg_list`, this would also work:

```python
self.results.cell_temperature = model(*arg_list, **kwargs)
```

note there is no need to cast the list to a tuple.

When fixing, we should add a test to ensure this value is passed properly.

Reply:
My mistake, I'll fix it.